src/main/java/com/example/digitalmenuapi/configuration/SecurityConfiguration.java

```diff
@@ -0,0 +1,73 @@
+package com.example.digitalmenuapi.configuration;
+
+import com.example.digitalmenuapi.service.AdminMenuItemsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * security configuration class for basic auth implementation and setup.
+ */
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration {
+
+  @Autowired
+  private AdminMenuItemsService adminMenuItemsService;
+
+  @Bean
+  public static PasswordEncoder passwordEncoder() {
+    return new BCryptPasswordEncoder();
+  }
+
+
+  /**
+   * created an InMemoryUserDetailsManager bean
+   * and I added an admin user with the username "admin" and password "staff123"
+   * with the role "ADMIN".
+   */
+  @Bean
+  public UserDetailsService userDetailsService() {
+    UserDetails admin = User.builder()
+            .username("admin")
+            .password(passwordEncoder().encode("staff123"))
```

Comment:
As a best practise these would be in a properties file so that these can be configured without having to rebuild / redeploy the application, https://www.baeldung.com/properties-with-spring